meta-lmp-base/recipes-bsp/u-boot/u-boot-fio-common.inc

```diff
@@ -51,4 +51,28 @@ do_deploy_append() {
         ln -sf ${PN}-config-${MACHINE}-${PV}-${PR} ${PN}-config-${MACHINE}
         ln -sf ${PN}-config-${MACHINE}-${PV}-${PR} ${PN}-config
     fi
+
+    SECONDARY_OFFSET=`grep 'define CONFIG_SECONDARY_BOOT_SECTOR_OFFSET' ${B}/${config}/u-boot.cfg | cut -d' ' -f 3`
+    if [ "${SECONDARY_OFFSET}" != "0" ]; then
+        bbnote "Building Secondary Image Table, firstSectorNumber = ${SECONDARY_OFFSET}"
+cat << EOF > sit_template
+#! /usr/bin/perl -w
+use strict;
+
+open(my \$out, '>:raw', 'sit.bin') or die "Unable to open: $!";
+print \$out pack("V", 0x0);	# chipnum
+print \$out pack("V", 0x0);	# driver type
+print \$out pack("V", 0x00112233);	# tag
+print \$out pack("V", %sectornum%);	# firstSectorNumber
+print \$out pack("V", 0x0);	# Sector count, not used
+close(\$out);
+
+EOF
+        sed -e s/%sectornum%/${SECONDARY_OFFSET}/g sit_template > sit_gen
+        chmod +x sit_gen
+        ./sit_gen
+        objcopy -I binary -O binary --pad-to 512 sit.bin
+        rm sit_gen sit_template
+        install -m 0644 sit.bin ${DEPLOYDIR}/sit-${MACHINE}.bin
+    fi
```

Comment:
Here we should generate sit as part of the do_compile function (via another append), and then deployed what it was built during that function to DEPLOYDIR. Kind of similar how it was done with imx-boot.

Reply:
we need to have u-boot compiled already, no?
or u-boot.cfg is generated in do_configure()?

```
SECONDARY_OFFSET=`grep 'define CONFIG_SECONDARY_BOOT_SECTOR_OFFSET' ${B}/${config}/u-boot.cfg | cut -d' ' -f 3`
```

Reply:
Sure, which is why it is safe to do it with a do_compile_append(), as it was done on imx-boot.

Configure can be just to configure u-boot itself, compile for building and generating all files required, and deploy just deploying to DEPLOYDIR.

This helps caching things correctly.